src/main/scala/wdl4s/RuntimeAttributes.scala

```diff
@@ -1,76 +1,91 @@
 package wdl4s
 
 import wdl4s.AstTools.{AstNodeName, EnhancedAstNode}
-import wdl4s.expression.NoFunctions
+import wdl4s.WdlExpression.ScopedLookupFunction
+import wdl4s.expression.{NoFunctions, WdlFunctions, WdlStandardLibraryFunctionsType}
+import wdl4s.parser.MemoryUnit
+import wdl4s.parser.WdlParser.{Ast, AstList, SyntaxError}
+import wdl4s.types.{WdlIntegerType, WdlStringType}
+import wdl4s.util.TryUtil
 import wdl4s.values._
-import wdl4s.parser.WdlParser.{Ast, AstList}
-import wdl4s.parser.MemorySize
 
 import scala.collection.JavaConverters._
 import scala.language.postfixOps
-import scalaz.Scalaz._
-import scalaz._
+import scala.util.{Failure, Success, Try}
 
-case class RuntimeAttributes(attrs: Map[String, Seq[String]])
+case class RuntimeAttributes(attrs: Map[String, WdlExpression]) {
+  import RuntimeAttributes._
+
+  def evaluate(lookup: ScopedLookupFunction, functions: WdlFunctions[WdlValue]): Map[String, Try[WdlValue]] = {
+    attrs map { case (k, v) =>
+      k -> v.evaluate(lookup, functions).flatMap(validateRuntimeValue(k, _))
+    }
+  }
+
+  private def validateRuntimeValue(key: String, value: WdlValue): Try[WdlValue] = {
+    key match {
+      case "memory" => validateMemoryValue(value).map(m => WdlString(m.toString))
+      case _ => Success(value)
+    }
+  }
+}
 
 object RuntimeAttributes {
-  def apply(ast: Ast): RuntimeAttributes = {
+  def apply(ast: Ast, declarations: Seq[Declaration]): RuntimeAttributes = {
     val asts = ast.findAsts(AstNodeName.Runtime)
     if (asts.size > 1) throw new UnsupportedOperationException("Only one runtime block may be defined per task")
-    val astList = asts.headOption map { _.getAttribute("map").asInstanceOf[AstList] }
-    val attrMap = astList map processRuntimeAttributes getOrElse Map.empty[String, Seq[String]]
-    attrMap.get("memory") flatMap { _.headOption } foreach validateMemory 
+    val kvPairAsts = asts.headOption.map(_.getAttribute("map").asInstanceOf[AstList].asScala.toVector.map(_.asInstanceOf[Ast]))
+    val runtimeAttributeMap = kvPairAsts match {
+      case Some(list) => list.map(ast => processRuntimeAttribute(ast, declarations)).toMap
+      case None => Map.empty[String, Try[WdlExpression]]
+    }
 
-    RuntimeAttributes(attrMap)
+    // .get below to throw the exception if validation didn't pass
+    TryUtil.sequenceMap(runtimeAttributeMap).map(RuntimeAttributes(_)).get
   }
 
-  /**
-    * Checks that a provided memory value is valid, if not throws an IllegalArgumentException
-    */
-  private def validateMemory(value: String): Unit = {
-    validateMemStringInGb (value) match {
-      case Failure(f) => throw new IllegalArgumentException("Invalid memory value specified:\n" + f.toList.mkString("\n"))
-      case Success(_) => ()
+  def validateMemoryValue(value: WdlValue): Try[MemorySize] = {
+    value match {
+      case i: WdlInteger => Success(MemorySize(i.value.toDouble, MemoryUnit.GB))
```

Comment:
The existing behavior is to default to GB and this is what the current pipeline expects.